python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -1321,6 +1319,7 @@ def get_best_checkpoint(
         best_path, best_metric = best_path_metrics[0]
         cloud_path = self._convert_local_to_cloud_path(best_path)
 
+        # TODO(matthewdeng): Figure out what to do here.
```

Comment:
Let's rename `NewExperimentAnalysis` to `ExperimentAnalysis` (this will affect a few other files), and rename `ExperimentAnalysis` to `LegacyExperimentAnalysis`.

Then, all usages of Checkpoint in `LegacyExperimentAnalysis` should be `air.Checkpoint as LegacyCheckpoint`.